plugins/cody-chat/plugin.xml

```diff
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <?eclipse version="3.4"?>
 <plugin>
+   <extension-point id="com.sourcegraph.cody.messageHandler" name="Cody Message Handler" schema="schema/com.sourcegraph.cody.messageHandler.exsd"/>
```

Comment:
Can you elaborate on the motivation for all this machinery? I would personally have implemented this functionality with an if/else condition. This XML config means we need to write quite a bit of code across multiple files to handle different scenarios.

Reply:
My instinct would have been to implement this like this instead
```
var message = ...
if (message instanceof NewFile) CodyCommands.newFile((NewFile) message)
else if (message instanceof InsertCodeAtPoint) CodyCommands.insertCodeAtPoint(...)
...
```

Reply:
I wanted to do that in a way that is idiomatic in eclipse. One benefit of doing it like this would be a possibility to later split up handlers affecting the ui and those only affecting the models into separate plugins within one feature. This would simplify testing later on. 

New handlers can be simply added from the plugin.xml UI. 
![image](https://github.com/user-attachments/assets/0346d075-4ace-4c9a-86c9-b8349270d2a6)
The `.exsd` file was auto-generated and probably won't be edited ever again.

All in all, this probably is an overkill for now. We can refactor that into simple if chain (or map) when we will be adding next handler.

Reply:
Makes sense. I think it's useful to have examples of these design patterns in the codebase even if it was overkill for this exact feature. This is exactly why it's great to have you on the team 🙏🏻